clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java

```diff
@@ -117,6 +118,17 @@ public int validBytes() {
         return downConvert(batches(), toMagic, firstOffset, time);
     }
 
+    /**
+     * Validates the header of the next batch and returns batch size. `buffer` should
+     * contain at least the header up to the size field. If buffer contains the magic
+     * byte, it is validated.
+     * @return next batch size including LOG_OVERHEAD
+     * @throws CorruptRecordException if record size or magic is invalid
+     */
+    public int nextBatchSize() {
```

Comment:
The notion of "next batch" is a little weird for `MemoryRecords` since it just represents a chunk of messages. Is `firstBatchSize` the intent?

Reply:
Good point, changed to `firstBatchSize`.